Source/WebCore/platform/graphics/FloatSegment.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <wtf/Vector.h>
+
+namespace WTF {
+class TextStream;
+}
+
+namespace WebCore {
+
+struct FloatSegment {
+    float begin { 0.f };
+    float end { 0.f };
+    bool operator==(const FloatSegment&) const = default;
+    float length() const { return end - begin; }
+    FloatSegment dilate(float length) { return { begin - length, end + length }; }
+};
+
+
+// Computes difference between a line segment `a` and list of line segments `bs` that are dilated by
+// dilationAmount.
+// The difference list is further filtered to only results > dilationAmount.
+//
+// Vector<FloatSegment> line { { 0, totalWidth } };
+// Vector<FloatSegment> diff = line - dilate(intersections, dilationAmount);
+// return filter(diff, [](auto&& s) { return s.length() > dilationAmount });
+// <=>
+// return differenceWithDilation({ 0, totalWidth }, intersections);
+inline Vector<FloatSegment> differenceWithDilation(FloatSegment a, Vector<FloatSegment>&& bs, float dilationAmount)
```

Comment:
I don't know if "dilation" is a term of art but there could be more explanation what this actually achieves, for example with some ascii art with lines and segments.

The pseudo-code is confusing, it is not clear `-` is an intersection operator for example.